### DISCUSSION TITLE
How do I install packages in Quarto's private Julia environment?

### DISCUSSION THREAD

**Discussion by jsimons8** [quarto-dev/quarto-cli#9587] [Q&A]
### Description

Hello!

I am trying out using the Julia engine as per instructions [here](https://quarto.org/docs/computations/julia.html#using-the-julia-engine) (new Quarto prerelease) and it's working great. May I ask how I can install packages though?

When I run the preview command of this minimal notebook
````qmd
---
title: "Plots Demo"
execute:
  echo: true
author: "Norah Jones"
date: "5/22/2021"
format:
  html:
    code-fold: true
engine: julia
---

### Parametric Plots

Plot function pair (x(u), y(u)). 
See @fig-parametric for an example.


```{julia}
#| label: fig-parametric
#| fig-cap: "Parametric Plots"

using Plots

plot(sin, 
     x->sin(2x), 
     0, 
     2π, 
     leg=false, 
     fill=(0,:lavender))
 
```

We can also try more here.
````

the output I am getting is

```
Running [1/1] at line 19:  using Plots, KalmanFilters
ERROR: Julia server returned error after receiving "run" command:
Failed to run notebook: /Users/me/Documents/ice/simulation-estimation/try.qmd
ERROR: EvaluationError: Encountered 1 error during evaluation

Error 1 of 1
@ /Users/jerome/Documents/ice/simulation-estimation/try.qmd:19
ArgumentError: Package KalmanFilters not found in current path.
- Run `import Pkg; Pkg.add("KalmanFilters")` to install the KalmanFilters package.
Stacktrace:
```

Is there a way to control the packages that this private Julia environment uses? Ideally, I would all install them from the command line in one go, since I have a collection I regularly use. Thank you!

**Comment by mcanouil** [comment]
See Julia documentation on environment: https://pkgdocs.julialang.org/v1/environments/

**Comment by cderv** [comment]
This is the same question as in 
- https://github.com/quarto-dev/quarto-cli/discussions/9551

What we document about environment is here: https://quarto.org/docs/computations/julia.html#engine-options

> For example, a notebook is run with --project=@. by default (the environment in the directory where the notebook is stored) but this could be overridden by setting

So this means you need to configure your environment local to the project you are rendering, or you need to setup the rendering to use another project. You can do that using the documented flag, or probably setting Julia environment variable [`JULIA_PROJECT` ](https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_PROJECT)

Julia engine is quite new, and it is not yet very well documented but we'll fix this 
- https://github.com/quarto-dev/quarto-cli/issues/9561

Thanks a lot for your feedback, it helps us improve our documentation

**Reply by dmi3kno** [reply]
when can `--project=@` fail to be the same as `--project=.` (assuming the qmd *is* in fact in the same directory as Project and Manifest)? I am developing a Julia package and I am not using Revise.jl (yet). Could it be that when environment indicated with `-project=@` the changes in the local directory may be ignored (and thus require the restart of REPL), while the changes made in the environment indicated by `-project=.` are picked up every time?

**Reply by mcanouil** [reply]
@dmi3kno would you mind opening a new discussion and put all your questions in it? You can reference the other threads.

All your questions are related to each other but right now it require us to reply to multiple threads and possibly to repeat/duplicate the answers.

Thanks you!

**Reply by dmi3kno** [reply]
I wish I knew what my question is. I will see if I can produce a reprex and diagnose what the issue is. Will definitely open a new discussion, once I understand what the problem is.

**Reply by cderv** [reply]
Just for context, regarding my answer you commented, as in our new doc, it is not ` --project=@` but ` --project=@.`. 

The trailing point is important here. See https://docs.julialang.org/en/v1/manual/environment-variables/index.html#JULIA_PROJECT
> If the variable is set to @. (note the trailing dot) then Julia tries to find a project directory that contains Project.toml or JuliaProject.toml file from the current directory and its parents.
